graylog2-web-interface/src/logic/errors/ReportedErrors.js

```diff
@@ -23,8 +28,20 @@ export const createReactError = (error: $PropertyType<ReactError, 'error'>, info
   info,
   type: ReactErrorType,
 });
-
 export const createUnauthorizedError = (error: $PropertyType<UnauthorizedError, 'error'>): UnauthorizedError => ({
   error,
   type: UnauthorizedErrorType,
 });
+export const createStreamPermissionError = (error: $PropertyType<StreamPermissionError, 'error'>): StreamPermissionError => ({
+  error,
+  type: StreamPermissionErrorType,
+});
+
+export const createFromFetchError = (error: FetchError) => {
+  switch (error.status) {
+    case 403:
+      return error?.additional?.body?.type === 'MissingStreamPermission' ? createStreamPermissionError(error) : createUnauthorizedError(error);
+    default:
+      throw Error(`Provided FetchError is not a valid ReportedError because status code ${error.status} is not supported`);
+  }
```

Comment:
This part is also the base to display the error message for 404 api requests, instead of redirecting to `/notfound`.
E.g. for the dashboard and saved search page, we just need to adjust the `ViewLoader`, once we have a custom `ReportedError`.